core/src/main/scala/kafka/server/FetchSession.scala

```diff
@@ -142,19 +141,20 @@ class CachedPartition(val topic: String,
       if (updateResponseData)
         localLogStartOffset = respData.logStartOffset
     }
-    if (respData.preferredReadReplica.isPresent) {
+    if (FetchResponse.preferredReadReplica(respData).isPresent) {
```

Comment:
Maybe we could have a isPreferredReplica method since there are at least two places where we just call isPresent on the result.